quark-test.c

```diff
@@ -436,6 +436,103 @@ t_fork_exec_exit(const struct test *t, struct quark_queue_attr *qa)
 	return (0);
 }
 
+static int
+t_process_iterator(const struct test *t, struct quark_queue_attr *qa)
+{
+	struct quark_queue		 qq;
+	struct quark_process_iter	 qi;
+	const struct quark_process	*qp;
+	int				 n_children = 3;
+	pid_t				 children[n_children];
+	struct args			*args;
+	size_t				 expected_len;
+	int				 i;
+	int				 j;
+	char				 cwd[PATH_MAX];
+
+	if (quark_queue_open(&qq, qa) != 0)
+		err(1, "quark_queue_open");
+
+	quark_process_iter_init(&qi, &qq);
+
+	for (i = 0; i < n_children; i++) {
+		children[i] = fork_exec_nop();
+		drain_for_pid(&qq, children[i]);
+	}
```

Comment:
```suggestion
	for (i = 0; i < (int)nitems(children); i++)
		children[i] = fork_exec_nop();
	for (i = 0; i < (int)nitems(children); i++)
		children[i] = drain_for_pid(&qq, children[i]);
```

Reply:
Small trick, use nitems() so you can cut n_children out.
Because of how aging works in quark, it makes more sense to fork all + drain all, otherwise each drain will have to age the event for 25ms, so this cuts 75ms->25ms